### PR TITLE
[WIP][SPARK-32093][BUILD] Add hadoop-ozone-filesystem jar.

### PR DESCRIPTION
**Pull request by bharatviswa504**
### What changes were proposed in this pull request?
Add ozone profile, so that ozone file system jar is packaged in spark tarball.

### Why are the changes needed?
This change is added so that spark can include hadoop-ozone-filesystem jar, and spark can run on top of ozone out of the box.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Created a new profile. It is a build change.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -202,6 +202,9 @@
     <arrow.version>0.15.1</arrow.version>
     <!-- org.fusesource.leveldbjni will be used except on arm64 platform. -->
     <leveldbjni.group>org.fusesource.leveldbjni</leveldbjni.group>
+    
+    <!-- ozone version -->
+    <ozone.version>0.5.0-beta</ozone.version>
```
Comment (dongjoon-hyun) [inline pom.xml:207]: Thank you for making a PR, but we usually don't add `-beta` dependency.

cc @dbtsai and @gatorsmile

File: pom.xml [hunk 2]
```diff
@@ -3059,13 +3062,28 @@
         <curator.version>2.7.1</curator.version>
         <commons-io.version>2.4</commons-io.version>
       </properties>
+     <dependencies>
```
Comment (dongjoon-hyun) [inline pom.xml:3065]: indentation?

File: pom.xml [hunk 3]
```diff
@@ -3059,13 +3062,28 @@
...
+      <dependencies>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-ozone-filesystem-lib-current</artifactId>
+          <scope>runtime</scope>
+          <version>${ozone.version}</version>
+        </dependency>
+      </dependencies>
     </profile>
-
```
Comment (dongjoon-hyun) [inline pom.xml:3068]: nit. Let's not touch this part.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
I'm wondering if you tried `--jars` or `--packages` first. That's the official way to use 3rd party library.

**Comment by dongjoon-hyun** [comment]
Gentle ping, @bharatviswa504 .

**Comment by srowen** [comment]
Why do we need this in Spark?

**Comment by dongjoon-hyun** [comment]
I also think like @srowen .

File: pom.xml (hunk 1 above)
Comment (bharatviswa504) [reply pom.xml:207]: Ozone will be GA next release. Discussion is happening in the community around this topic.

Ozone has been scale tested with 1 billion objects and also it is tested with TPCSDS Hive and Spark.

**Comment by bharatviswa504** [comment]
The reason for doing this is similar to the idea of spark packaging cloud connector jars, so that spark will be out of the box can work with them. (And the user does not require to copy jars to spark executor classpath)

Similar support we want to add for ozone also. Right now to make spark work with ozone, we are using a --files and then running spark jobs. If spark executor has ozone jar included, we shall not require this step.

Purpose of this Jira is to include an ozone file system jar in spark release tarball, in this way spark can out of box work with an ozone cluster when used with ozone paths.

**Comment by srowen** [comment]
Does it work if a user app just includes it?
Can it go in hadoop-cloud if it must be packaged with Spark in some contexts -- why a new profile?
but what is Ozone, a file system from the Hadoop project, and who / how many people would need this?

**Comment by bharatviswa504** [comment]
> Does it work if a user app just includes it?
Yes, and also it requires configuration required to connect to the ozone cluster. They can be included in core-site or in a new file ozone-site.xml.

> Can it go in hadoop-cloud if it must be packaged with Spark in some contexts -- why a new profile?
Initially, I have thought of a new profile, later used the existing Hadoop profile only.
I have updated the PR description to reflect the same.

> but what is Ozone, a file system from the Hadoop project, and who / how many people would need this?
Right now it is a sub-project of Hadoop.
Ozone is a scalable, redundant, and distributed object store for Hadoop, and also it provides Hadoop compatible FileSystem where spark/hive applications can run on it.

Link to Ozone website
https://hadoop.apache.org/ozone/

**Comment by srowen** [comment]
OK, if it's something that logically belongs as part of the cluster so that not every app needs to include it, I can see trying to build it into a Spark distro. It wouldn't be enabled by default. It could go in hadoop-cloud though this isn't really to help use cloud object stores, though then again maybe this is meant to be a cloud-based object store in its own right?
In any event not sure it makes sense to add to the pom here if it's still pretty beta / dev? Anyone can add this to a downstream Spark build already.

File: pom.xml (hunk 1 above)
Comment (dongjoon-hyun) [reply pom.xml:207]: What is the next `Ozone will be GA next release`? Could you close this and reopen after that with the official version?

Comment (dongjoon-hyun) [reply pom.xml:207]: I marked this PR as `[WIP]`. Please remove that when the PR is ready.

Comment (bharatviswa504) [reply pom.xml:207]: Ozone's next release is 0.6.0. 
Mostly it will be released before August 2020.

**Comment by bharatviswa504** [comment]
> t not every app needs to include it, I can see trying to build it into a Spark distro. It wouldn't be enabled by default. It could go in hadoop-cloud though this isn't really to help use cloud object stores, though then again maybe this is meant to be a cloud-based object store in its own right?
> In any event not sure it makes sense to add to the pom here if it's still pretty beta / dev? Anyone can add this to a downstream Spark build already.

I will modify this PR, once ozone GA version is released.

This change is to make ozone support out of box with spark tarball distro. So, that spark distro's can contain required ozone jar. This is the main intention of this.

Ozone is an on-prem object store. So, I have added this main pom.xml, as hadoop-cloud is specifically for cloud stores.

**Pull request closed without merging**